### PR TITLE
Check null for reject sword sc

### PR DESCRIPTION
**Pull request by kaninhot004**
* **Addressed Issue(s)**: -

* **Server Mode**: Renewal

* **Description of Pull Request**: Fix Reject Sword made map server crash.

How to reproduce
1. Made custom script
```
-	script	Test	-1,{
end;

OnPCDieEvent:
killmonsterall strcharinfo(3);
end;
}
```
2. Spawn Stalker monster
3. Hit it with Wind Cutter and damage reflect till character die
4. Map server crash.

### CODE REVIEW

File: src/map/battle.cpp [hunk 1]
```diff
@@ -6069,9 +6069,11 @@ static void battle_calc_weapon_final_atk_modifiers(struct Damage* wd, struct blo
 	{
 		ATK_RATER(wd->damage, 50)
 		status_fix_damage(target,src,wd->damage,clif_damage(target,src,gettick(),0,0,wd->damage,0,DMG_NORMAL,0,false),ST_REJECTSWORD);
-		clif_skill_nodamage(target,target,ST_REJECTSWORD,tsc->data[SC_REJECTSWORD]->val1,1);
-		if( --(tsc->data[SC_REJECTSWORD]->val3) <= 0 )
-			status_change_end(target, SC_REJECTSWORD, INVALID_TIMER);
+		if (tsc->data[SC_REJECTSWORD] != NULL) {
```
Comment (aleos89) [inline src/map/battle.cpp:6072]:
```suggestion
		if (tsc->data[SC_REJECTSWORD] != nullptr) {
```

**Review by aleos89 — APPROVED** [review]

**Comment by aleos89** [comment]
I just looked around the code a bit more for any more instances of SC_REJECTSWORD and in doing so I noticed that we already check for the nullptr value above where you added another. I tried to reproduce the crash but I can't.

Can you provide a bit more details into your crash scenario? Like what items you are using, specific monsters (ID), etc. Thanks!

**Review by null6385 — APPROVED** [review]

**Comment by Lemongrass3110** [comment]
Unable to reproduce the crash (without your fix).
As @aleos89 already pointed out this status is already checked for nullpointer 2 lines above.

**Comment by Lemongrass3110** [comment]
> > Unable to reproduce the crash (without your fix). As @aleos89 already pointed out this status is already checked for nullpointer 2 lines above.
> 
> Have you tried my custom script that contain killmonsterall? I think that might lead to issue.

@kaninhot004 yes. Exactly the same script.

**Comment by CairoLee** [comment]
seems like the `tsc->data[SC_REJECTSWORD]` is a null pointer

**Comment by aleos89** [comment]
I did took another jab at this and I was still unable to reproduce it (using your script and same mobs). It's obvious from your screen shot that the SC is null, but it seems something is out of order. It doesn't seem right that we have the null check above but then 2 calls later, which don't impede the SC, results in it becoming null.

Are you testing this against a clean rAthena or do you have significant src changes?

**Comment by vstumpf** [comment]
This is a valid problem, but very rare.
The important thing is the script. 

1) Stalker Gertie casts Reject Sword
2) LK uses Wind Cutter, damages Gertie
3) Gertie takes damage, reflects damage back to LK
4) LK dies to reflect
5) Script runs, killing all monsters (and clearing their sc)
6) We try to lower the SC_REJECTSWORD val1, which fails. 

```
Breakpoint 2, pc_dead (sd=sd@entry=0x55555d046830, src=src@entry=0x55556fa668bc) at pc.cpp:9184
9184    {
(gdb) bt
#0  pc_dead (sd=sd@entry=0x55555d046830, src=src@entry=0x55556fa668bc) at pc.cpp:9184
#1  0x0000555555872681 in status_damage (src=src@entry=0x55556fa668bc, target=target@entry=0x55555d046830,
    dhp=<optimized out>, dsp=dsp@entry=0, dap=dap@entry=0, walkdelay=0, flag=0, skill_id=390) at status.cpp:1484
#2  0x00005555555d67f3 in status_damage (skill_id=390, flag=0, walkdelay=<optimized out>, dsp=0, dhp=<optimized out>,
    target=0x55555d046830, src=0x55556fa668bc) at status.hpp:3148
#3  status_fix_damage (skill_id=390, walkdelay=<optimized out>, hp=<optimized out>, target=0x55555d046830,
    src=0x55556fa668bc) at status.hpp:3148
#4  battle_calc_weapon_final_atk_modifiers (wd=wd@entry=0x7fffffffdc30, src=src@entry=0x55555d046830,
    target=target@entry=0x55556fa668bc, skill_id=skill_id@entry=2005, skill_lv=5) at battle.cpp:6102
#5  0x00005555555d8610 in battle_calc_weapon_attack (src=0x55555d046830, target=0x55556fa668bc,
    skill_id=<optimized out>, skill_lv=<optimized out>, wflag=<optimized out>) at battle.cpp:6759

(gdb) c
Continuing.

Program received signal SIGSEGV, Segmentation fault.
0x00005555555d680e in battle_calc_weapon_final_atk_modifiers (wd=wd@entry=0x7fffffffdc30,
    src=src@entry=0x55555d046830, target=target@entry=0x55556fa668bc, skill_id=skill_id@entry=2005, skill_lv=5)
    at battle.cpp:6103
6103                    clif_skill_nodamage(target,target,ST_REJECTSWORD,tsc->data[SC_REJECTSWORD]->val1,1);
(gdb) bt
#0  0x00005555555d680e in battle_calc_weapon_final_atk_modifiers (wd=wd@entry=0x7fffffffdc30,
    src=src@entry=0x55555d046830, target=target@entry=0x55556fa668bc, skill_id=skill_id@entry=2005, skill_lv=5)
    at battle.cpp:6103
#1  0x00005555555d8610 in battle_calc_weapon_attack (src=0x55555d046830, target=0x55556fa668bc,
    skill_id=<optimized out>, skill_lv=<optimized out>, wflag=<optimized out>) at battle.cpp:6759
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**